### PR TITLE
Remove `github.com/packethost/pkg/env` dependency

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

Remove `github.com/packethost/pkg/env` dependency and remove unused code. While I tried to stay away from as much refactoring as possible, there is some in here that was needed in order to remove the `github.com/packethost/pkg/env` dependency. Other than this, code was moved around to decrease the cognitive load of understanding this code base and prepare us for a more holistic refactor that will include the deprecation of `github.com/packethost/dhcp4-go` and `github.com/packethost/pkg` entirely.

## Why is this needed

Fixes: #

## How Has This Been Tested?

Will be doing some manual testing with the Helm chart. Will update here with the results, once complete.

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: cmd/boots/env.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		}
+		v4 := ip.IP.To4()
+		if v4 == nil || !v4.IsGlobalUnicast() {
+			continue
+		}
+
+		return v4
+	}
+	err = errors.New("unable to auto-detect public IPv4")
+	panic(err)
```
Comment (chrisdoherty4) [inline cmd/boots/env.go:84]:
```suggestion
	panic("unable to auto-detect public IPv4")
```

File: cmd/boots/env.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+func mustPublicSyslogIPv4() net.IP {
+	if s, ok := os.LookupEnv("PUBLIC_SYSLOG_IP"); ok {
+		if a := net.ParseIP(s).To4(); a != nil {
+			return a
+		}
+		err := errors.New("PUBLIC_SYSLOG_IP must be an IPv4 address")
+		panic(err)
```
Comment (chrisdoherty4) [inline cmd/boots/env.go:93]:
```suggestion
		panic("PUBLIC_SYSLOG_IP must be an IPv4 address")
```

File: dhcp/server/server.go [hunk 3]
```diff
@@ -66,22 +62,91 @@ type dhcpHandler struct {
...
+	slice := strings.Split(macs, ",")
+	if len(slice) == 0 {
+		return nil
+	}
+
+	ignore := map[string]struct{}{}
+	for _, oui := range slice {
+		_, err := net.ParseMAC(oui + ":00:00:00")
+		if err != nil {
+			panic(errors.Errorf("invalid oui in TINK_IGNORED_OUIS oui=%s", oui))
```
Comment (chrisdoherty4) [inline dhcp/server/server.go:99]: Should this be panicking?

File: cmd/boots/env.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	return 0
+}
+
+func mustPublicIPv4() net.IP {
+	if s, ok := os.LookupEnv("PUBLIC_IP"); ok {
+		if a := net.ParseIP(s).To4(); a != nil {
+			return a
+		}
+		err := errors.New("PUBLIC_IP must be an IPv4 address")
+		panic(err)
```
Comment (chrisdoherty4) [inline cmd/boots/env.go:64]:
```suggestion
		panic("PUBLIC_IP must be an IPv4 address")
```

File: cmd/boots/env.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+// getIntEnv parses given environment variable as an int, or returns the default if the environment variable is empty/unset.
+// Int will panic if it fails to parse the value.
+func getIntEnv(name string, def ...int) int {
+	if v := os.Getenv(name); v != "" {
+		i, err := strconv.Atoi(v)
+		if err != nil {
+			err = errors.Wrap(err, "failed to parse int from env var")
+			panic(err)
```
Comment (chrisdoherty4) [inline cmd/boots/env.go:48]: Should this be panicking?

File: dhcp/server/server.go [hunk 6]
```diff
@@ -66,22 +62,91 @@ type dhcpHandler struct {
...
+
+	slice := strings.Split(ips, ",")
+	if len(slice) == 0 {
+		return nil
+	}
+
+	ignore := map[string]struct{}{}
+	for _, ip := range slice {
+		if net.ParseIP(ip) == nil {
+			panic(errors.Errorf("invalid ip address in TINK_IGNORED_GIS ip=%s", ip))
```
Comment (chrisdoherty4) [inline dhcp/server/server.go:132]: Should this be panicking?

**Review by chrisdoherty4 — APPROVED** [review]
I suspect, given the intent to refactor more holistically, the comments aren't really worth addressing.

**Comment by jacobweinstock** [comment]
> I suspect, given the intent to refactor more holistically, the comments aren't really worth addressing.

Yeah, exactly. I have another [PR](https://github.com/tinkerbell/boots/pull/326) thats in the works to clean up all of this. Thanks for the review!

**Pull request merged**